### ISSUE TITLE
Report train wrecks in methods to simplify their usage

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#1881]
We should report warnings for so-called train wrecks that occur within a method multiple times and propose to store them into variables.

Example:
```csharp
public void DoSomething()
{
    A.B.C.D.E.F.G.h1();
    A.B.C.D.E.F.G.h2();
    A.B.C.D.E.F.G.h3();
    A.B.C.D.E.F.G.h4();
    A.B.C.D.E.F.G.h5();
}
```

Such code is difficult to read. It would be much easier to read in case the code would look like:
```csharp
public void DoSomething()
{
    var g = A.B.C.D.E.F.G;

    g.h1();
    g.h2();
    g.h3();
    g.h4();
    g.h5();
}
```

**Comment by RalfKoban** [comment]
Consider following sitations:
- Statement -> Invocation -> SimpleMember -> SimpleMember -> SimpleMember -> SimpleMember -> IdentifierName
- Argument -> SimpleMember -> SimpleMember -> SimpleMember -> IdentifierName
- AssignmentExpression  -> SimpleMember -> SimpleMember -> SimpleMember -> SimpleMember -> IdentifierName
- BinaryExpression -> SimpleMember -> SimpleMember -> SimpleMember -> SimpleMember -> IdentifierName
